res/layout/instant_onboarding_activity.xml

```diff
@@ -0,0 +1,131 @@
+      android:layout_height="match_parent"
+      android:layout_marginTop="20dp"
+      android:orientation="vertical">
+
+    <LinearLayout android:layout_width="match_parent"
+                  android:layout_height="wrap_content"
+                  android:layout_marginLeft="16dp"
+                  android:layout_marginRight="16dp"
+                  android:orientation="horizontal">
+
+      <ImageView android:id="@+id/avatar"
+                 android:layout_width="64dp"
+                 android:layout_height="64dp"
+                 android:contentDescription="@string/pref_profile_photo"
+                 android:transitionName="avatar"/>
+
+      <com.google.android.material.textfield.TextInputLayout
+          android:id="@+id/name"
+          android:layout_width="match_parent"
+          android:layout_height="wrap_content"
+          android:layout_marginLeft="10dp">
+
+        <org.thoughtcrime.securesms.components.emoji.EmojiEditText
+            android:id="@+id/name_text"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:hint="@string/pref_your_name"
+            android:inputType="textCapWords" />
+      </com.google.android.material.textfield.TextInputLayout>
+
+    </LinearLayout>
+
+    <TextView
+        android:id="@+id/information_label"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:layout_marginLeft="16dp"
+        android:layout_marginRight="16dp"
+        android:layout_marginTop="16dp"
+        android:gravity="start"
+        android:text="@string/set_name_and_avatar_explain"
+        android:textColor="@color/gray50" />
+
+    <TextView
+        android:id="@+id/privacy_policy_button"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_gravity="start"
+        android:layout_marginLeft="16dp"
+        android:layout_marginRight="16dp"
+        android:layout_marginTop="16dp"
+        android:text="@string/instant_onboarding_agree_default"
+        android:textColor="@color/delta_accent"
+        android:textSize="16sp"/>
+
+    <Button
+        style="@style/ButtonPrimary"
+        android:id="@+id/signup_button"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:paddingLeft="16dp"
+        android:paddingRight="16dp"
+        android:layout_marginLeft="16dp"
+        android:layout_marginRight="16dp"
+        android:layout_marginTop="20dp"
+        android:layout_marginBottom="16dp"
+        android:text="@string/instant_onboarding_create"/>
+
+    <androidx.legacy.widget.Space
+        android:layout_weight="2"
+        android:layout_width="match_parent"
+        android:layout_height="0dp"/>
+
+    <androidx.constraintlayout.widget.ConstraintLayout
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content">
```

Comment:
why not also use a `LinearLayout` here?

unless there are good reasons, in general, `LinearLayout` is easier maintainable and understandable, esp. by non-pro android-devs :)

no blocker, though

Reply:
it was LinearLayout at the beginning, but had to switch to ConstraintLayout to make the labels wrap if they are too long, in LinearLayout they just push everything out of the screen if ex. the first item is too long, but maybe similar results can be achieved using LinearLayout + weights, this was just the first working solution I found when searching for this problem

Reply:
i see, and i agree that these layouts are tricky sometimes and one is happy to find sth. working :) and with the tuned down layout, even the wrapping looks okay now

so, i am fine leaving it as a `ConstraintLayout`,  might anyways be, that this parts gets changes soon

but a <Space> might be better instead of empty text, semantic-wise, screen reader etc. , see suggestion below